DSpace/dspace-angular
# Replace dso page edit buttons with a menu

## Description
This PR removes the separate buttons from the various DSpace object pages, such as the edit button and the versioning button, and puts them in a menu which will be rendered as buttons.

To add new buttons to the edit pages, these can be added to the menus in the `DSOEditMenuResolver`, similar to the general `MenuResolver`. 

Every top level menu will be rendered as a button. When this menu has no child menus, clicking the button will perform the action defined in the menu section in the resolver. When the menu has child menus, clicking the button will show a dropdown containing the actions defined in the child menu sections.

Furthermore, because the version button on the item page can be disabled, the menu functionality was expanded with a disabled option. Support for this disabled property was added to all other existing menus, but is currently not in use.

## Instructions for Reviewers
The changes were made in such a way that the UI remains the same for the different dspace object pages. 
To verify this, you can do the following:
* As an anonymous user or a user that does not have edit rights:
  * Navigate to some community, collection and item pages
  * Verify that you do not see any edit buttons  
* As an admin user or a user authorised to edit dspace objects:
  * Navigate to some community, collection and item pages
  * Verify that you can see the edit button
  * Verify that you can see the item version button on the item page and that it behaves as before   

## Checklist
